pkgs/build-support/setup-hooks/writable-tmpdir-as-home.sh

```diff
@@ -0,0 +1,15 @@
+# shellcheck shell=bash
+# This setup hook set the HOME environment variable to a writable directory.
+
+export HOME
+
+writableTmpDirAsHome () {
+    if [ ! -w "$HOME" ]; then
+        HOME="$NIX_BUILD_TOP/.home"
```

Comment:
I worry that this will [break](https://github.com/nixos/nixpkgs/issues/189691) when run under nix-shell. Can we use something other than $NIX_BUILD_TOP please? (Or define a new env var for where the build actually happens, which is also valid in nix-shell?)

Reply:
Well, break nix-shell _and_ use of the `keepBuildTree` setup hook, as per the linked issue above.